### PR TITLE
MAINT - Removed all the StringLiteralDuplication suppressions and addressed issues

### PR DESCRIPTION
**Pull request by blen-desta**
- Changed CommonDataTypeVerifier to a class with static methods
- Removed all the StringLiteralDuplication suppressions and addressed issues

### CODE REVIEW

File: ctk/common/src/main/kotlin/org/codice/compliance/utils/decorators/IdpRedirectResponseDecorator.kt [hunk 1]
```diff
@@ -50,17 +50,6 @@ internal constructor(response: IdpRedirectResponse) : IdpRedirectResponse(respon
         paramMap[SIGNATURE]
     }
 
-    val extraUrlParameters: Boolean by lazy {
```
Comment (blen-desta) [inline ctk/common/src/main/kotlin/org/codice/compliance/utils/decorators/IdpRedirectResponseDecorator.kt]: Removed this unused method.

Comment (coyotesqrl) [reply ctk/common/src/main/kotlin/org/codice/compliance/utils/decorators/IdpRedirectResponseDecorator.kt]: I'd say even if we keep it, there are better ways of determining its value.

File: ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CommonDataTypeVerifier.kt [hunk 2]
```diff
@@ -31,156 +30,170 @@ import java.net.URI
...
+        private const val NO_HYPHEN_MESSAGE = "No hyphen was found in the date value of"
+
+        var ids = mutableListOf<String>()
+
+        /** 1.3 Common Data Types **/
+        fun verifyCommonDataType(samlDom: Node) {
+            ids = mutableListOf()
+            var i = samlDom.childNodes.length - 1
+
+            while (i >= 0) {
```
Comment (coyotesqrl) [inline ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CommonDataTypeVerifier.kt]: ✏️ A ranged loop would be cleaner:
`for (i in samlDom.childNodes.length - 1 downTo 0) {`
you can remove the var and get rid of the decrement on line 59 with that.

File: ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CommonDataTypeVerifier.kt [hunk 3]
```diff
@@ -31,156 +30,170 @@ import java.net.URI
...
+                    verifyIdValues(child, null)
+
+                if (child.hasChildNodes())
+                    verifyCommonDataType(child)
+                i -= 1
+            }
+        }
+
+        /** 1.3.1 String Values **/
+        @SuppressWarnings("StringLiteralDuplication")
```
Comment (coyotesqrl) [inline ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CommonDataTypeVerifier.kt]: ❗️ Should be kotlin's `@Suppress` annotation.

**Pull request merged**